### PR TITLE
Check kept on new password entered while changing password

### LINKED ISSUE #98: Password length to be checked when changing password as well
**Describe the bug**
The check on the length of the password is there while signing up but not for changing the password.

Related to Quality Assurance issue #101 

**To Reproduce**
1. Go to http://systers-mentorship-dev.eu-central-1.elasticbeanstalk.com/
2. Create your user by registering at /register
3. Try to change the password with another one which has a length lesser than 8
4. It get's changed

**Expected behavior**
It should have returned an error saying the min length of a password is 8.

**Screenshots**
-N.A.-

**Desktop (please complete the following information):**
 - OS: Linux
 - Browser Chrome

### LINKED ISSUE #99: Add validation to Change Password API request data
## Description
As a developer,
I need password validation to be the same as in POST /register,
so that I can avoid users violating business rules.

Solves issue #98 

### Update
- [x] Add a validation function to app/api/validations/user.py file
- [x] Add tests to prove that the validation function is working properly. The test file can be called `test_validation_change_password.py` inside folder  ` tests/users`
- [x] Use this validation function in `ChangePassword` class, funtion `put()` from file `app/api/resources/user.py` to validate the request data sent by the client is being verified
- [x] **Warning!** make sure to strip strings before checking for the length!! fix this in the registration validation and add tests!

## Definition of Done
- [x] All of the required items are completed.
- [x] PR should be sent to the `gsoc18-code` branch.
- [x] Approval by 1 mentor.

## Estimation
2 hours

### PR DESCRIPTION
**Pull request by sammy1997**
### Description
Added a validation function to check if the new password entered while password change is valid.

Fixes #98 
Fixes #99 

### Type of Change:
- Code
- Quality Assurance

### How Has This Been Tested?
-N.A-

### Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] My changes generate no new warnings 
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by sammy1997** [comment]
@isabelcosta The PR for #98 . Please label this and connect it to the issue on Zenhub!
@m-murad @Dilu9218

**Review by isabelcosta — CHANGES_REQUESTED** [review]
@sammy1997 it looks good to me, but it would be great if you could add at least 2 tests one for the minimum and other for the maximum length, you can follow the examples in this file: https://github.com/systers/mentorship-backend/blob/gsoc18-code/tests/users/test_validation_request_data.py  

And you could also follow this project commit style message: https://github.com/systers/mentorship-backend/wiki/Commit-Message-Style-Guide, in this case this is sort of a fix

**Comment by sammy1997** [comment]
@isabelcosta Changes made!

File: tests/users/test_validation_request_data.py [hunk 1]
```diff
@@ -184,6 +184,24 @@ def test_user_registration_request_data_password_superior_to_limit(self):
...
+                           "characters and shorter than {max_limit} characters.".format(min_limit=PASSWORD_MIN_LENGTH-1,
+                                                                                        max_limit=PASSWORD_MAX_LENGTH+1)}
+        random_generated_password = "".join(secure_random.choice(ascii_lowercase) for x in range(PASSWORD_MIN_LENGTH-1))
+        actual_result = validate_new_password(random_generated_password)
+        self.assertEqual(expected_result, actual_result)
+
+    def test_user_password_change_request_password_superior_to_limit(self):
+        secure_random = SystemRandom()
+        expected_result = {"message": "The password field has to be longer than {min_limit} "
+                           "characters and shorter than {max_limit} characters.".format(min_limit=PASSWORD_MIN_LENGTH-1,
```
Comment (Monal5031) [inline tests/users/test_validation_request_data.py]: MAX_LENGTH :p

File: tests/users/test_validation_request_data.py [hunk 2]
```diff
@@ -184,6 +184,24 @@ def test_user_registration_request_data_password_superior_to_limit(self):
...
+                                                                                        max_limit=PASSWORD_MAX_LENGTH+1)}
+        random_generated_password = "".join(secure_random.choice(ascii_lowercase) for x in range(PASSWORD_MIN_LENGTH+1))
```
Comment (Monal5031) [inline tests/users/test_validation_request_data.py]: MAX_LENGTH :p

Comment (Monal5031) [inline tests/users/test_validation_request_data.py]: `PASSWORD_MAX_LENGTH+1` :p

**Review by Monal5031 — CHANGES_REQUESTED** [review]

File: tests/users/test_validation_request_data.py (hunk 2 above)
Comment (sammy1997) [reply tests/users/test_validation_request_data.py]: Thanks @Monal5031 !

**Comment by isabelcosta** [comment]
@sammy1997  in the commit message you don't need "Type: Fix. Check ..." you just need "fix: check ..."

File: app/api/validations/user.py [hunk 3]
```diff
@@ -71,3 +71,13 @@ def validate_resend_email_request_data(data):
         return {"message": "Your email is invalid."}
 
     return {}
+
+
+def validate_new_password(password):
```
Comment (isabelcosta) [inline app/api/validations/user.py]: Can you add here an extra validation to see in the `current_password` and `new_password` are in data object? Because if a user doesn't send that when you call this function assuming that the `new_password` is being sent, you'll get an exception because you're trying to access the `new_password` in data when it does not exist. You can see the example on `validate_resend_email_request_data`.

File: app/api/resources/user.py [hunk 4]
```diff
@@ -120,6 +121,9 @@ def put(cls):
         """
         user_id = get_jwt_identity()
         data = request.json
+        is_valid = validate_new_password(data['new_password'])
```
Comment (isabelcosta) [inline app/api/resources/user.py]: instead of receiving the data[`new_password`] you could receive data and do all the verifications because that will give an exception if `new_password` isn't there @sammy1997

**Review by isabelcosta — CHANGES_REQUESTED** [review]

**Review by isabelcosta — APPROVED** [review]

**Review by Dilu9218 — APPROVED** [review]

**Review by Monal5031 — APPROVED** [review]

**Review by r00pal — APPROVED** [review]
